src/compiler/checker.ts

```diff
@@ -1065,16 +1065,23 @@ namespace ts {
             return diagnostic;
         }
 
-        function addErrorOrSuggestion(isError: boolean, diagnostic: DiagnosticWithLocation) {
-            if (isError) {
+        function addErrorOrWarningOrSuggestion(category: DiagnosticCategory, diagnostic: DiagnosticWithLocation) {
+            if (category === DiagnosticCategory.Error) {
                 diagnostics.add(diagnostic);
             }
             else {
-                suggestionDiagnostics.add({ ...diagnostic, category: DiagnosticCategory.Suggestion });
+                suggestionDiagnostics.add({ ...diagnostic, category });
             }
         }
+
         function errorOrSuggestion(isError: boolean, location: Node, message: DiagnosticMessage | DiagnosticMessageChain, arg0?: string | number, arg1?: string | number, arg2?: string | number, arg3?: string | number): void {
-            addErrorOrSuggestion(isError, "message" in message ? createDiagnosticForNode(location, message, arg0, arg1, arg2, arg3) : createDiagnosticForNodeFromMessageChain(location, message)); // eslint-disable-line no-in-operator
+            const category = isError ? DiagnosticCategory.Error : DiagnosticCategory.Suggestion;
+            addErrorOrWarningOrSuggestion(category, "message" in message ? createDiagnosticForNode(location, message, arg0, arg1, arg2, arg3) : createDiagnosticForNodeFromMessageChain(location, message)); // eslint-disable-line no-in-operator
+        }
+
+        function wraningOrSuggestion(isWraning: boolean, location: Node, message: DiagnosticMessage | DiagnosticMessageChain, arg0?: string | number, arg1?: string | number, arg2?: string | number, arg3?: string | number): void {
+            const category = isWraning ? DiagnosticCategory.Warning : DiagnosticCategory.Suggestion;
+            addErrorOrWarningOrSuggestion(category, "message" in message ? createDiagnosticForNode(location, message, arg0, arg1, arg2, arg3) : createDiagnosticForNodeFromMessageChain(location, message)); // eslint-disable-line no-in-operator

```

Comment:
```suggestion
        function warningOrSuggestion(isWarning: boolean, location: Node, message: DiagnosticMessage | DiagnosticMessageChain, arg0?: string | number, arg1?: string | number, arg2?: string | number, arg3?: string | number): void {
            const category = isWarning ? DiagnosticCategory.Warning : DiagnosticCategory.Suggestion;
            addErrorOrWarningOrSuggestion(category, "message" in message ? createDiagnosticForNode(location, message, arg0, arg1, arg2, arg3) : createDiagnosticForNodeFromMessageChain(location, message)); // eslint-disable-line no-in-operator
```

Reply:
@sandersn VS Code does not include suggestion level diagnostics in the problems view, only errors, warnings, and information messages